embed.h

```diff
@@ -101,15 +101,17 @@
 #define ck_entersub_args_list(a)	Perl_ck_entersub_args_list(aTHX_ a)
 #define ck_entersub_args_proto(a,b,c)	Perl_ck_entersub_args_proto(aTHX_ a,b,c)
 #define ck_entersub_args_proto_or_list(a,b,c)	Perl_ck_entersub_args_proto_or_list(aTHX_ a,b,c)
-#ifndef MULTIPLICITY
-#define ck_warner		Perl_ck_warner
-#define ck_warner_d		Perl_ck_warner_d
+#if !defined(MULTIPLICITY) || defined(PERL_CORE)
+#define ck_warner(a,...)	Perl_ck_warner(aTHX_ a,__VA_ARGS__)
+#endif
+#if !defined(MULTIPLICITY) || defined(PERL_CORE)
```

Comment:
Why the `!defined(MULTIPLICITY)`? Shouldn't that definition work regardless?

Reply:
It probably should but maybe that will be surprising to XS authors? Up until now, it's been available in third-party XS but only when `!defined(MULTIPLICITY)`. It's a silly reason, but I didn't want to break anything. If we want to revisit that we could as a separate idea.

Reply:
> Up until now, it's been available in third-party XS but only when !defined(MULTIPLICITY). It's a silly reason, but I didn't want to break anything. If we want to revisit that we could as a separate idea.

But if we don't make it accessible to `!defined(MULTIPLICITY)`, none of this is actually useful.

Reply:
> Up until now, it's been available in third-party XS but only when !defined(MULTIPLICITY). It's a silly reason, but I didn't want to break anything. If we want to revisit that we could as a separate idea.

But if we don't make it available to `!defined(MULTIPLICITY)`, none of this is actually useful so what would be the point?

Reply:
This commit makes them visible within core, finally. Lots of code in perl core currently does `Perl_warn(aTHX_ ...)` because it cannot `warn(...)`. This commit would fix that.
A future commit could also remove the `defined(MULTIPLICTY) &&` part of this condition to make them available elsewhere as well. That would need more testing though to see if it breaks any CPAN modules.
I wanted to take the first step of making it available within core first so we can use it in our own code.

Reply:
Fair, I'll leave it for a follow-up